File: tests/fixtures/kotlin/basic_class.kt
package com.example.models

import java.util.UUID

/**
 * A basic user class with primary constructor.
 */
class User(
    val id: UUID,
    val name: String,
    var email: String
) {
    private var loginCount: Int = 0

    fun displayName(): String {
        return "$name <$email>"
    }

    fun incrementLogin() {
        loginCount++
    }
}

File: tests/fixtures/scala/basic_class.scala
package com.example.models

import java.util.UUID

/**
 * A basic user class with primary constructor.
 */
class User(val id: UUID, var name: String, email: String) {
  private val createdAt: Long = System.currentTimeMillis()

  def greet(): String = s"Hello, $name"

  private def validate(): Boolean = name.nonEmpty

  override def toString: String = s"User($id, $name)"
}
